### DISCUSSION TITLE
Configure webhooks with OAuth2 Token

### DISCUSSION THREAD

**Discussion by ishita-chakraborty** [netbox-community/netbox#9324] [Q&A]
I would like to configure a webhook in netbox. The webhook is secured with OAuth2 with client credentials grant type, implying a 2 step authentication, where we can get the token from a outh2 url and use the token in an Authorization header. Is it possible to do this in netbox. Also if the above is not possible, i was trying to pass a static username and password in the header and that too is not working. Can someone provide a picture of how the additional header row be populated in the webhook configuration form?
![image](https://user-images.githubusercontent.com/84405532/167519892-b326578d-a598-4207-956f-cffeb6fe539d.png)

**Comment by WagoL** [comment]
@ishita-chakraborty How did deal with this OAuth2 case?

**Comment by mauricehome** [comment]
I think that it is of importance that the proposed OAuth2 functionality would actually be supported by Netbox. 
Would it be possible to implement this is in a Netbox plugin ?

**Comment by WagoL** [comment]
I have a script that modifies the webhook with a new bearer token. And a cron job.

**Comment by mauricehome** [comment]
Hi, so it is custom after all.
In fact it is all about sending a patch request to {{url}}/api/extras/webhooks/<webhookid> with a payload like:
{
    "additional_headers" : "Authorization: Bearer <token>"
}
on a regular basis, with <token> the token that was most recently retrieved.
Thanks!
